sway-core/src/ir_generation/function.rs

```diff
@@ -1995,9 +2138,8 @@ impl<'eng> FnCompiler<'eng> {
 
         // We must compile the RHS before checking for shadowing, as it will still be in the
         // previous scope.
-        let body_deterministically_aborts = body.deterministically_aborts(self.engines.de(), false);
         let init_val = self.compile_expression_to_value(context, md_mgr, body)?;
```

Comment:
Why not simply call `return_on_termination_or_extract!` here?

Reply:
`return_on_termination_or_extract!` potentially returns a value of type `TerminatorValue`. The function `compile_var_decl` needs to return a value of type `Option<TerminatorValue>`, so we can't use the macro here. This is also the only place were an `Option<TerminatorValue>` needs to be returned, so I didn't think it was worth it to generalize the macro or to add a new one.